vms/avm/txs/txstest/builder.go

```diff
@@ -0,0 +1,233 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txstest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/vms/avm/config"
+	"github.com/ava-labs/avalanchego/vms/avm/state"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/signer"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+type Builder struct {
+	backend *utxos
+	ctx     *builder.Context
+}
+
+func New(
+	codec codec.Manager,
+	ctx *snow.Context,
+	cfg *config.Config,
+	feeAssetID ids.ID,
+	state state.State,
+) *Builder {
+	backend := newBackend(ctx, state, ctx.SharedMemory, codec)
+	return &Builder{
+		backend: backend,
+		ctx:     newContext(ctx, cfg, feeAssetID),
+	}
+}
+
+func (b *Builder) CreateAssetTx(
+	name, symbol string,
+	denomination byte,
+	initialStates map[uint32][]verify.State,
+	kc *secp256k1fx.Keychain,
+	changeAddr ids.ShortID,
+) (*txs.Tx, error) {
+	xBuilder, xSigner := b.builders(kc)
+
+	utx, err := xBuilder.NewCreateAssetTx(
+		name,
+		symbol,
+		denomination,
+		initialStates,
+		common.WithChangeOwner(&secp256k1fx.OutputOwners{
+			Threshold: 1,
+			Addrs:     []ids.ShortID{changeAddr},
+		}),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed building base tx: %w", err)
+	}
```

Comment:
```suggestion
	if utx, err := xBuilder.NewCreateAssetTx(
		name,
		symbol,
		denomination,
		initialStates,
		common.WithChangeOwner(&secp256k1fx.OutputOwners{
			Threshold: 1,
			Addrs:     []ids.ShortID{changeAddr},
		}),
	); err != nil {
		return nil, fmt.Errorf("failed building base tx: %w", err)
	}
```

Reply:
Can't be done, utx is used by the signed right below